test/database/persistence/survey_persistence_test.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:survey_flutter_ic/database/dto/survey_dto.dart';
+import 'package:survey_flutter_ic/database/persistence/survey_persistence.dart';
+
+import '../../mocks/generate_mocks.mocks.dart';
+
+void main() {
+  group('SurveyPersistence', () {
+    late MockBox mockBox;
+    late SurveyPersistence surveyPersistence;
+
+    final List<SurveyDto> surveysDto = <SurveyDto>[
+      const SurveyDto(
+        id: 'id',
+        title: 'title',
+        description: 'description',
+        isActive: true,
+        coverImageUrl: 'coverImageUrl',
+        largeCoverImageUrl: 'largeCoverImageUrl',
+        createdAt: 'createdAt',
+        surveyType: 'surveyType',
+      ),
+    ];
+
+    setUp(() {
+      mockBox = MockBox();
+      surveyPersistence = SurveyPersistenceImpl(mockBox);
+    });
+
+    test(
+        'When calling Surveys form cache, it emits the corresponding SurveyDto',
+        () async {
+      when(mockBox.get('surveyKey', defaultValue: [])).thenReturn(surveysDto);
+
+      final result = await surveyPersistence.surveys();
+
+      expect(result, surveysDto);
+      verify(mockBox.get('surveyKey', defaultValue: []));
```

Comment:
This `verify` seems redundant, since we added the stubbing to return `surveysDto`, so when the method call doesn't match the `expect` should be failed.

```
when(mockBox.get('surveyKey', defaultValue: [])).thenReturn(surveysDto);
```

Reply:
Oh, yeah, your right @hoangnguyen92dn 👍 It's redundant with the mock scope.